### DISCUSSION TITLE
Support programmatic content seeding

### DISCUSSION THREAD

**Discussion by ggoodman** [jameslittle230/stork#134] [Ideas]
In [this issue comment](https://github.com/jameslittle230/stork/issues/11#issuecomment-761902301) I asked if Stork might consider a different mode of index generation where the content doesn't need to be serialized to disk. 

The use-case I have in mind is for a sort of SSR / SSG hybrid that I hope to eventually allow running in environments where there will be no filesystem. In that scenario, I was wondering if there was a programmatic interface that could be explored to seed the index without any filesystem involved.

**Comment by styfle** [comment]
I think a good solution would be reading `stdin` for the toml contents and utilizing the [`file.contents`](https://stork-search.net/docs/config-ref/) property to pass the data directly.

Something like:

```
cat example.toml | stork --build-stdin
```

**Reply by ggoodman** [reply]
Wow, missed that! Sounds very plausible!

**Reply by jameslittle230** [reply]
Following up here -- this wasn't possible before, but 1.1.0 will include #145, which will allow you to pipe in a TOML configuration into the `stork --build` command!
